src/shared/find-esp.c

```diff
@@ -302,10 +303,9 @@ static int verify_fsroot_dir(
                  * user we can stat() /boot/efi/, and we can stat()/enumerate /boot/. But we cannot look into
                  * /boot/efi/, and in particular not use /boot/efi/../ – hence this work-around.) */
 
-                if (path_equal(path, "/"))
-                        goto success;
-
                 r = path_extract_directory(path, &parent);
+                if (r == -EDESTADDRREQ)
+                        goto success;
```

Comment:
this hunk probably should be its own commit?

Reply:
not addressed?

Reply:
Why should this be its own commit? Before switching to fd based path handling, the paths were absolute and we needed the path_equal("/") check. After switching to fd based path handling, the paths are relative and so we change the check to take this into account. I'm not sure why this should be in a separate commit hence.